test/powershell/Host/ConsoleHost.Tests.ps1

```diff
@@ -711,8 +711,53 @@ namespace StackTest {
             $longPipeName = [string]::new("A", 200)
 
             "`$pid" | & $powershell -CustomPipeName $longPipeName -c -
-            # 64 is the ExitCode for BadCommandLineParameter
-            $LASTEXITCODE | Should -Be 64
+            $LASTEXITCODE | Should -Be $ExitCodeBadCommandLineParameter
+        }
+    }
+
+    Context "ApartmentState tests" -Tag Slow {
+
+        It "Should be able to set apartment state to: <apartment>" -Skip:(!$IsWindows) -TestCases @(
+            @{ apartment = "STA"; switch = "-sta" }
+            @{ apartment = "MTA"; switch = "-mta" }
+        ) {
+            param ($apartment, $switch)
+
+            & $powershell $switch -noprofile -command "[System.Threading.Thread]::CurrentThread.ApartmentState" | Should -BeExactly $apartment
+        }
+
+        It "WPF will work when using -sta" -Skip:(!$IsWindows) {
+            $script = [scriptblock]::Create({
+                add-type -AssemblyName presentationframework
+
+                $xaml = [xml]@"
+                <Window
+                    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+                    x:Name="Window" Title="Initial Window" WindowStartupLocation = "CenterScreen"
+                    Width = "400" Height = "300" ShowInTaskbar = "True">
+                </Window>
+"@
+
+                $reader = [System.Xml.XmlNodeReader]::new($xaml)
+                $Window = [System.Windows.Markup.XamlReader]::Load($reader)
+                # This will throw an exception if MTA
+                $Window.Show()
+                $Window.Close()
+            })
+
+            & $powershell -sta -noprofile -command $script
```

Comment:
If we make STA the default, then we should also have a test case where we do not specify the switch at all and WPF works out of the box

Reply:
Can we directly look an apartment in default thread?

Reply:
@iSazonov yes, with change to STA as default, I'll add a test to ensure that is the case.